### PR TITLE
Added SofortPayment Doctrine Entity

### PR DESCRIPTION
**Pull request by JeroenDeDauw**
For https://phabricator.wikimedia.org/T167886

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
I will add some extra tests to make sure the thing is behaving as expected in a bit

**Comment by JeroenDeDauw** [comment]
Oh and, the getters :D

File: src/Entities/Payments/SofortPayment.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Entities\Payments;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @since 6.0
+ *
+ * @ORM\Table(name="payment_sofort")
```
Comment (gbirke) [inline src/Entities/Payments/SofortPayment.php]: I vaguely remember agreeing on each domain (donation, membership) having its own payment data table, so this should probably be `donation_payment_sofort`. Even when memberships are never paid with sofort, at least our database table names are consistent.

Comment (JeroenDeDauw) [inline src/Entities/Payments/SofortPayment.php]: If the persistence is truly private to each BC, then this does not matter. You cannot have a naming conflict with something in another "namespace" (in this case persistence mechanism). I suppose having the prefix anyway might make getting there easier though, so will add it.

Comment (JeroenDeDauw) [inline src/Entities/Payments/SofortPayment.php]: Prefix definitely makes the current intent clearer

Comment (JeroenDeDauw) [inline src/Entities/Payments/SofortPayment.php]: I was thinking about how to namespace the payment class earlier.

    WMDE\Fundraising\DonationContext\SomeDatabaseStuff\SofortPayment

**Comment by JeroenDeDauw** [comment]
> Is there any reason why you pass in the donation id manually instead of using Doctrine associations? 

Not knowing that there is such a thing as Doctrine associations :)

**Comment by JeroenDeDauw** [comment]
Sounds plausible, but as I do not know this stuff, and do not really get the motivation for using it, it'd be awesome if you could demonstrate.

**Pull request merged**